### PR TITLE
Feature/docs

### PR DESCRIPTION
**Pull request by anguera5**
Added autodocs functionality and implemented docstrings from enviPath class

### CODE REVIEW

File: docs/.DS_Store
Comment (lorsbach) [inline docs/.DS_Store:1]: Please remove and put .DS_Store into gitignore

File: docs/source/conf.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Configuration file for the Sphinx documentation builder.
+#
+# For the full list of built-in configuration values, see the documentation:
+# https://www.sphinx-doc.org/en/master/usage/configuration.html
+
+# -- Project information -----------------------------------------------------
+# https://www.sphinx-doc.org/en/master/usage/configuration.html#project-information
+
+project = 'enviPath-python'
+copyright = '2023, Albert Anguera Sempere'
```
Comment (lorsbach) [inline docs/source/conf.py:10]: Should be adjusted to "2023, enviPath UG & Co. KG"

File: enviPath_python/enviPath.py [hunk 2]
```diff
@@ -15,12 +15,15 @@
...
 from collections.abc import Iterable
+from typing import Union, List
 
 from requests import Session
 from requests.adapters import HTTPAdapter
 from requests.exceptions import *
 
 from enviPath_python.objects import *
+from enviPath_python.objects import SimpleRule, SequentialCompositeRule, ParallelCompositeRule, User, Package, Compound, \
+    Pathway, Reaction, Scenario, Setting, Rule
```
Comment (lorsbach) [inline enviPath_python/enviPath.py:26]: Whats the purpose of the duplicated import? Either import only whats needed or all

File: enviPath_python/enviPath.py [hunk 3]
```diff
@@ -68,6 +86,13 @@ def who_am_i(self) -> User:
         return User(self.requester, **user_data)
 
     def search(self, term: str, packages: Union['Package', List['Package']]):
+        """
+        Function designed to perform a search on an enviPath session.
+
+        :param term: the term with which the search wants to be performed
+        :param packages: the packages where the search wants to be performed
+        :return: the matching objects ids
```
Comment (lorsbach) [inline enviPath_python/enviPath.py:94]: return is dict of object identifiers such as "reaction" to a list of lazy initialized objects

**Review by lorsbach — COMMENTED** [review]
See inline comments

File: docs/.DS_Store
Comment (anguera5) [reply docs/.DS_Store:1]: Fixed

**Review by lorsbach — APPROVED** [review]

**Pull request merged**